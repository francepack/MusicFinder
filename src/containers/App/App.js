import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import Loading from '../../components/Loading/Loading'
import BandInput from '../BandInput/BandInput'
import BandInfo from '../BandInfo/BandInfo'
import Events from '../Events/Events'
import { Route, withRouter, Link } from 'react-router-dom'

export class App extends Component {
  constructor(props) {
    super(props)
    this.state = {
      error: '',
      loading: false
    }
  }

  setError = (message) => {
    this.setState({ error: message })
  }

  setLoading = () => {
    this.setState({ loading: !this.state.loading })
  }
  
  render() {
    const { error } = this.state
    return (
      <div className='App'>
        <h1>
          <Link to='/'>
            MusicFinder
          </Link>
        </h1>
        <BandInput setError={this.setError} setLoading={this.setLoading} />
        {!this.props.band && 
          <div className='instructions'>
            <p>Type in the name of your favorite band above.</p>
            <p>Submit to find similar bands and descriptions of that band.</p>
            <p>Click on any result to search for events that may be of interest.</p>
          </div>
        }
        {error && error}
        {this.state.loading && <Loading />}
        <Route exact path='/' />
        <Route path='/Loading' />
        <Route path='/band-info' component={() => <BandInfo setError={this.setError} setLoading={this.setLoading} />}/>
        <Route path='/events' component={() => <Events /> }/>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  band: state.band
})

App.propTypes = {
  band: PropTypes.string
}

export default withRouter(connect(mapStateToProps, null)(App))