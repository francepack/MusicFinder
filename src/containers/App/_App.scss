body {
  background-color: $darkBlue;
  font-family: $standardFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  margin: 0;
  padding: 0;
  text-align: center;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

a {
  @include links;
}

h1 {
  background-color: $greenBlue;
  border-bottom: 6px solid $wood;
  color: $platinum;
  font-family: $fancyFont;
  font-size: 320%;
  margin-top: 0; 
  padding: 40px 0;
}

.instructions {
  color: $darkPlat;
  cursor: default;
  margin-top: 50px;
  padding: 0 20px;
}

.instructions p {
  margin-bottom: 20px;
  line-height: 28px;
}