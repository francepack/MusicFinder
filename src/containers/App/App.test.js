import React from 'react'
import { shallow } from 'enzyme'
import { App, mapStateToProps } from './App'


describe('App', () => {
  let wrapper
  const mockTags = ['cool', 'rock', 'alternative']
  beforeEach(() => {
    wrapper = shallow(
      <App {...mockTags} />
    )
  })
  describe('app', () => {
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
    it('should match snapshot when error is present', () => {
      wrapper.setState({ error: 'Fetch call failed' })
      expect(wrapper).toMatchSnapshot()
    })
    it('should have default state', () => {
      const expected = {
        error: '',
        loading: false
      }
      expect(wrapper.state()).toEqual(expected)
    })
  })

  describe('setError', () => {
    it('should set state error', () => {
      const initialState = {
        error: '',
        loading: false
      }
      const expected = {
        error: 'Error! Error!',
        loading: false
      }
      wrapper.setState(initialState)
      expect(wrapper.state()).toEqual(initialState)
      const mockError = 'Error! Error!'
      wrapper.instance().setError(mockError)
      expect(wrapper.state()).toEqual(expected)
    })
  })

  describe('setLoading', () => {
    it('should change state loading', () => {
      const initialState = {
        error: '',
        loading: false
      }
      const expected = {
        error: '',
        loading: true
      }
      wrapper.setState(initialState)
      expect(wrapper.state()).toEqual(initialState)
      wrapper.instance().setLoading()
      expect(wrapper.state()).toEqual(expected)
    })
  })
  
  describe('mapStateToProps', () => {
    it('should return an object with similarBands and tags', () => {
      const state = {
        band: '',
        similarBands: [],
        events: [],
        tags: []
      }
      const expected = {
        band: ''
      }
      const result = mapStateToProps(state)
      expect(result).toEqual(expected)
    })
  })
})