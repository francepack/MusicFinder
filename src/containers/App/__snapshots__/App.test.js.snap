// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App app should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    0="cool"
    1="rock"
    2="alternative"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          <Link
            to="/"
          >
            MusicFinder
          </Link>
        </h1>,
        <withRouter(Connect(BandInput))
          setError={[Function]}
          setLoading={[Function]}
        />,
        "",
        <Instructions />,
        false,
        <Route
          exact={true}
          path="/"
        />,
        <Route
          path="/Loading"
        />,
        <Route
          component={[Function]}
          path="/band-info"
        />,
        <Route
          component={[Function]}
          path="/events"
        />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            to="/"
          >
            MusicFinder
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "MusicFinder",
            "to": "/",
          },
          "ref": null,
          "rendered": "MusicFinder",
          "type": [Function],
        },
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "setError": [Function],
          "setLoading": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/Loading",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/band-info",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/events",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            <Link
              to="/"
            >
              MusicFinder
            </Link>
          </h1>,
          <withRouter(Connect(BandInput))
            setError={[Function]}
            setLoading={[Function]}
          />,
          "",
          <Instructions />,
          false,
          <Route
            exact={true}
            path="/"
          />,
          <Route
            path="/Loading"
          />,
          <Route
            component={[Function]}
            path="/band-info"
          />,
          <Route
            component={[Function]}
            path="/events"
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              to="/"
            >
              MusicFinder
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "MusicFinder",
              "to": "/",
            },
            "ref": null,
            "rendered": "MusicFinder",
            "type": [Function],
          },
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "setError": [Function],
            "setLoading": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/Loading",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/band-info",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/events",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`App app should match snapshot when error is present 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    0="cool"
    1="rock"
    2="alternative"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          <Link
            to="/"
          >
            MusicFinder
          </Link>
        </h1>,
        <withRouter(Connect(BandInput))
          setError={[Function]}
          setLoading={[Function]}
        />,
        "Fetch call failed",
        <Instructions />,
        false,
        <Route
          exact={true}
          path="/"
        />,
        <Route
          path="/Loading"
        />,
        <Route
          component={[Function]}
          path="/band-info"
        />,
        <Route
          component={[Function]}
          path="/events"
        />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            to="/"
          >
            MusicFinder
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "MusicFinder",
            "to": "/",
          },
          "ref": null,
          "rendered": "MusicFinder",
          "type": [Function],
        },
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "setError": [Function],
          "setLoading": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "Fetch call failed",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/Loading",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/band-info",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/events",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            <Link
              to="/"
            >
              MusicFinder
            </Link>
          </h1>,
          <withRouter(Connect(BandInput))
            setError={[Function]}
            setLoading={[Function]}
          />,
          "Fetch call failed",
          <Instructions />,
          false,
          <Route
            exact={true}
            path="/"
          />,
          <Route
            path="/Loading"
          />,
          <Route
            component={[Function]}
            path="/band-info"
          />,
          <Route
            component={[Function]}
            path="/events"
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              to="/"
            >
              MusicFinder
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "MusicFinder",
              "to": "/",
            },
            "ref": null,
            "rendered": "MusicFinder",
            "type": [Function],
          },
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "setError": [Function],
            "setLoading": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "Fetch call failed",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/Loading",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/band-info",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/events",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
